ballerina/tests/from_xml_test.bal

```diff
@@ -0,0 +1,628 @@
+            city: "Colombo",
+            country: "Sri Lanka"
+        },
+        codes: {
+            item: [4, 8, 9]
+        },
+        'xmlns\:ns0: "http://sample.com/test",
+        status: "online"
+    };
+    BookStore4 actual = check fromXml(payload);
+    test:assertEquals(actual, expected, msg = "testToRecordWithNamespaces result incorrect");
+}
+
+type BookStore5 record {
+    xml storeName;
+    int postalCode;
+    boolean isOpen;
+    xml address;
+    xml codes;
+    @Attribute
+    string status;
+    @Attribute
+    string 'xmlns\:ns0;
+};
+
+type Address5 record {
+    string street;
+    string city;
+    string country;
+};
+
+type Codes5 record {
+    int[] item;
+};
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testRecordToXml5() returns error? {
+    xml payload = xml `<BookStore5 status="online" xmlns:ns0="http://sample.com/test">
+                            <storeName>foo</storeName>
+                            <postalCode>94</postalCode>
+                            <isOpen>true</isOpen>
+                            <address>
+                                <street>Galle Road</street>
+                                <city>Colombo</city>
+                                <country>Sri Lanka</country>
+                            </address>
+                            <codes>
+                                <item>4</item>
+                                <item>8</item>
+                                <item>9</item>
+                            </codes>
+                        </BookStore5>
+                        <!-- some comment -->
+                        <?doc document="book.doc"?>`;
+    BookStore5 expected = {
+        storeName: xml `foo`,
+        postalCode: 94,
+        isOpen: true,
+        address: xml `<street>Galle Road</street><city>Colombo</city><country>Sri Lanka</country>`,
+        codes: xml `<item>4</item><item>8</item><item>9</item>`,
+        'xmlns\:ns0: "http://sample.com/test",
+        status: "online"
+    };
+    BookStore5 actual = check fromXml(payload);
+    test:assertEquals(actual, expected, msg = "testRecordToXml5 result incorrect");
+}
+
+xml xmValue = xml `<Bill xmlns="example.com" attr="attr-val" xmlns:ns="ns.com" ns:attr="ns-attr-val">
+                <PurchesedItems>
+                    <PLine><ItemCode>223345</ItemCode><Count>10</Count></PLine>
+                    <PLine><ItemCode>223300</ItemCode><Count>7</Count></PLine>
+                    <PLine><ItemCode discount="22%">200777</ItemCode><Count>7</Count></PLine>
+                </PurchesedItems>
+                <Address xmlns="">
+                    <StreetAddress>20, Palm grove, Colombo 3</StreetAddress>
+                    <City>Colombo</City>
+                    <Zip>300</Zip>
+                    <Country>LK</Country>
+                </Address>
+              </Bill>`;
+
+type Bill record {
+    PurchesedItems1 PurchesedItems;
+    Address10 Address;
+    string 'xmlns?;
```

Comment:
this should come as namespace annotation

Reply:
No, `xmlns` is not a prefix. So, it can come as an attribute annotation.

If we add it as namespace annotation in the Open API definition, 
```
components:
  schemas:
    Bill:
      type: object
      properties:
        foo:
          type: object
      xml:
        prefix: xmlns
        namespace: 'http://www.w3.org/2001/XMLSchema-instance'  
```
The equivalent XML format is
```
<xmlns: Bill xmlns:xmlns="http://www.w3.org/2001/XMLSchema-instance">
	<foo>
	</foo>
</xmlns:Bill>
```